### PR TITLE
feat/add eve production domain

### PR DESCRIPTION
**Pull request by aarmoa**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Added EVE Exchange production URLs for HTTP and websocket connections.
Included the required changes in the connector to support domains. This allowed to keep the current URLs as a staging domain (eve_staging).
Changed README.md file to include EVE Exchange in the list with its logo

The PR also includes a change in the configuration of the accepted spot connectors list for the "exchange" field in the basic Pydantic config map class for strategies. This change was required to allow the use of subdomain connectors. Affected strategies:
- Avellaneda
- Cross Exchange Market Making

**Tests performed by the developer**:
Tested by running a PMM strategy again the staging domain for ETH-USDT.
All unit tests passing.

**Tips for QA testing**:
Check that the staging version of the connector can still be used with any spot strategy.
Validate that the production endpoints work as expected.

[sc-31577]

### CODE REVIEW

**Comment by rxlxrxsx** [comment]
QA Test Run: https://hummingbot.testrail.io/index.php?/runs/view/1831

**Review by petioptrv — APPROVED** [review]
Changes LGTM!

**Comment by gerald-hb** [comment]
Test update:
We can now login to eve exchange prod and asset has been transferred. We will resume testing this PR

**Review by rxlxrxsx — APPROVED** [review]
Tested and confirmed working by QA.

- Connected API keys (valid and invalid key) and confirmed working as expected
- Confirmed wallet and exchange balances are reflecting correct
- Created a pure_mm strategy and went through initial prompts successfully
- Started the strategy without issues, confirmed orders are created and cancelled based on config
- Confirmed order book prices is in sync with the exchange
- Trade information are correct based on history command output
- Compared trades in client and exchange, no discrepancies found
- Confirmed hanging order/cancellation and multiple order levels works as expected
- Tested on other supported strategies as well and confirmed working
- Successfully built a docker image and created a docker instance

**Comment by aarmoa** [comment]
@CoinAlpha/qa-team I have pushed a change that modifies the basic Pycharm configuration class for strategies. Now the list of accepted spot connectors is populated in a different way. I have tested the change locally already with a normal installation. Could you check again please if you can create and execute an Avellaneda strategy with a subdomain connector (like eve staging)?

File: hummingbot/client/config/strategy_config_data_types.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    def validate_exchange_trading_pair(cls, v: str, values: Dict):
+        exchange = values.get("exchange")
+        ret = validate_market_trading_pair(exchange, v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+
+class BaseTradingStrategyMakerTakerConfigMap(BaseStrategyConfigMap):
+    maker_market: str = Field(
```
Comment (petioptrv) [inline hummingbot/client/config/strategy_config_data_types.py:84]: This strategy base should also be adapted with the change

File: hummingbot/client/config/strategy_config_data_types.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                value="Exchanges",  # noqa: F821
+                names={e: e for e in AllConnectorSettings.get_connector_settings().keys()},
+                type=str,
+            )
+        if field.name == "taker_market_trading_pair":
+            cls.__fields__["taker_market"].type_ = ClientConfigEnum(  # rebuild the exchanges enum
+                value="Exchanges",  # noqa: F821
+                names={e: e for e in AllConnectorSettings.get_connector_settings().keys()},
+                type=str,
+            )
```
Comment (petioptrv) [inline hummingbot/client/config/strategy_config_data_types.py:153]: These should also be removed with the new approach

File: hummingbot/client/config/strategy_config_data_types.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        return v
+
+
+class BaseTradingStrategyConfigMap(BaseStrategyConfigMap):
+    exchange: ClientConfigEnum(  # rebuild the exchanges enum
+        value="Exchanges",  # noqa: F821
+        names={e: e for e in
+               sorted([connector_setting.name for connector_setting
+                       in AllConnectorSettings.get_connector_settings().values()
+                       if connector_setting.type is ConnectorType.Exchange] + settings.PAPER_TRADE_EXCHANGES)},
```
Comment (petioptrv) [inline hummingbot/client/config/strategy_config_data_types.py:38]: Why not use `AllConnectorSettings. get_exchange_names` instead?

**Review by petioptrv — CHANGES_REQUESTED** [review]
A request to update the bi-exchange strategy base, but I would also agree to putting that off to another issue if LQS is not currently using XEMM.

**Comment by gerald-hb** [comment]
Posting it here first as I am not sure if this is an issue or expected behavior.

We could create and start an Avellaneda strategy but we are not able to make any trades. It is stuck in `calculating volatility, estimating orderbook liquidity, no trades tick`. I have tested it in different market like ETH-USDT and XRP-USDT in Eve_staging but still no luck. This issue also happen with kucointestnet
image
image

When running pure market making strategy, we were able to capture trades. 
[logs_kucointestnet.log](https://github.com/CoinAlpha/hummingbot/files/9590710/logs_kucointestnet.log)
[logs_avetest2.log](https://github.com/CoinAlpha/hummingbot/files/9590711/logs_avetest2.log)

File: hummingbot/client/config/strategy_config_data_types.py (hunk 1 above)
Comment (aarmoa) [reply hummingbot/client/config/strategy_config_data_types.py]: I applied the change

File: hummingbot/client/config/strategy_config_data_types.py (hunk 2 above)
Comment (aarmoa) [reply hummingbot/client/config/strategy_config_data_types.py]: I removed it for both fields

File: hummingbot/client/config/strategy_config_data_types.py (hunk 3 above)
Comment (aarmoa) [reply hummingbot/client/config/strategy_config_data_types.py]: Good point. I made the change.

**Comment by aarmoa** [comment]
@CoinAlpha/qa-team to address the comments made by Petio I had to change the Pydantic configuration for the maker and taker connectors of XEMM strategy. Please validate that the strategy can be configured correctly (the autocomplete for the taker market is broken, but it is also like that in the development branch. It was broken by one merge done by the community, and it is not fixed in this PR)

**Comment by aarmoa** [comment]
@CoinAlpha/qa-team regarding the avellaneda strategy not starting to trade please make sure you are using connectors and trading pairs that do have trades happening in the market. Avellaneda needs the public trades from the exchange to produce the trading intensity.

File: hummingbot/client/config/strategy_config_data_types.py [hunk 4]
```diff
@@ -81,15 +77,23 @@ def validate_exchange_trading_pair(cls, v: str, values: Dict):
 
 
 class BaseTradingStrategyMakerTakerConfigMap(BaseStrategyConfigMap):
-    maker_market: str = Field(
+    maker_market: ClientConfigEnum(  # rebuild the exchanges enum
```
Comment (petioptrv) [inline hummingbot/client/config/strategy_config_data_types.py:80]:
```suggestion
    maker_market: ClientConfigEnum(
```

File: hummingbot/client/config/strategy_config_data_types.py [hunk 5]
```diff
@@ -81,15 +77,23 @@ def validate_exchange_trading_pair(cls, v: str, values: Dict):
...
+    ) = Field(
         default=...,
         description="The name of the maker exchange connector.",
         client_data=ClientFieldData(
             prompt=lambda mi: "Enter your maker spot connector",
             prompt_on_new=True,
         ),
     )
-    taker_market: str = Field(
+    taker_market: ClientConfigEnum(  # rebuild the exchanges enum
```
Comment (petioptrv) [inline hummingbot/client/config/strategy_config_data_types.py:92]:
```suggestion
    taker_market: ClientConfigEnum(
```

File: hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making_config_map_pydantic.py [hunk 6]
```diff
@@ -359,15 +359,14 @@ class CrossExchangeMarketMakingConfigMap(BaseTradingStrategyMakerTakerConfigMap)
...
-    maker_market: str = Field(
-        default=...,
-        description="The name of the maker exchange connector.",
-        client_data=ClientFieldData(
-            prompt=lambda mi: "Enter your maker spot connector (Exchange)",
-            prompt_on_new=True,
-        ),
-    )
-    taker_market: str = Field(
+    taker_market: ClientConfigEnum(  # rebuild the exchanges enum
```
Comment (petioptrv) [inline hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making_config_map_pydantic.py:362]:
```suggestion
    taker_market: ClientConfigEnum(
```

**Review by petioptrv — CHANGES_REQUESTED** [review]
Left a couple of suggestions to remove inaccurate in-line comments, but the rest looks good to me

**Review by gerald-hb — APPROVED** [review]
LGTM! (retest)
* Connected API keys (valid and invalid key) and confirmed working as expected
* Confirmed wallet and exchange balances are reflecting correct
* Created a pure_mm strategy and went through initial prompts successfully
* Started the strategy without issues, confirmed orders are created and cancelled based on config
* Confirmed that XEMM and Avellaneda was able to start successfully
* Confirmed order book prices is in sync with the exchange
* Successfully built a docker image and created a docker instance

**Review by petioptrv — APPROVED** [review]
The changes LGTM!

**Comment by aarmoa** [comment]
Closing the PR after creating it in the foundation repo https://github.com/hummingbot/hummingbot/pull/5722

**Pull request closed without merging**